### ISSUE TITLE
Escaping # in metric instance name in telegraf.conf

### ISSUE THREAD

**Issue by Aleyho** [influxdata/telegraf#1546]
### Steps to reproduce:

While calling a inputs.win_perf_counters.object, Object type process, I'm running into a problem with having an instance name with a # as part of the instance. On a particular server I have two of the same processes running but I only need to monitor the 2nd.  As I label the instance "process#1" for example naturally it comments out the rest of the line. If I attempt to escape it with "process\#1" this also breaks the entire process and no results from telegraf are collected. 
### Expected behavior:

\ should escape the # and allow "process\#1" as an instance name for a process
### Actual behavior:

"process\#1" as an instance name causes Telegraf to relay no data
### Additional info:

  [[inputs.win_perf_counters.object]]
      ObjectName = "Process"
      Counters = ["% Privileged Time", "% User Time"]
      Instances = ["Ax32Serv\#1"]
      Measurement = "win_process"

**Comment by sparrc** [comment]
what happens if you double-escape it?

```
[[inputs.win_perf_counters.object]]
ObjectName = "Process"
Counters = ["% Privileged Time", "% User Time"]
Instances = ["Ax32Serv\\#1"]
Measurement = "win_process"
```

**Comment by sparrc** [comment]
or also try it with single quotes, with no escaping:

```
[[inputs.win_perf_counters.object]]
  ObjectName = "Process"
  Counters = ["% Privileged Time", "% User Time"]
  Instances = ['Ax32Serv#1']
  Measurement = "win_process"
```

**Comment by Aleyho** [comment]
Incidentally those are the first two things I tried, but unfortunately neither worked.

**Comment by sparrc** [comment]
that's unfortunate....might be a bug in the toml parsing library we use then 😕, don't really have time now to track it down but any PRs or additional debugging are greatly appreciated.

**Comment by Aleyho** [comment]
No worries, I'll see what I can do.

**Comment by dmitrysdm** [comment]
I think i have the same problem, follow config does n't working
```
  [[inputs.win_perf_counters.object]]
    # Example query where the Instance portion must be removed to get data back, such as from the Memory object.
    ObjectName = "NVIDIA GPU"
    Counters = ["% GPU Usage", "% GPU Memory Usage", "Temperature (ºC)"]
    Instances = ["*"] # Use 6 x - to remove the Instance bit from the query.
    Measurement = "win_nvidia_gpu"
    #IncludeTotal=false #Set to true to include _Total instance when querying for all (*).
```
maybe this happens because my nvidia card instance also have # in the name!
`#0 Quadro 2000 (id=1, NVAPI ID=3840)`

**Comment by danielnelson** [comment]
I believe this has been fixed by #1827.
